### DISCUSSION TITLE
Ability to have different actions on the physical buttons based on the current page

### DISCUSSION THREAD

**Discussion by TeDeVPrime** [Blackymas/NSPanel_HA_Blueprint#1189] [Ideas]
Hello,

is it possible to have the physical buttons have different actions based on the page or entity page we are in?

For example... let's say i open a media player entity..... for UX reasons... i would like to press the right button to turn the volume up and the left button to turn the volume down.....

If i am inside a light entity.... the long press right button sets the brightness higher.... if i long press the left button it turns the brightness down....

if i am in the sensors page.... pressing the right button it will go to the next page... pressing the left button it will go to the previous page.....

If someone rings the doorbell... i would like to call the doorbell entity (as a switch) in front in the nspanel and by pressing ANY button (left or right) i would like to open the door.

that way even a blind guy can use this switch...

"oh the doorbell rings? ok... just press any button to open the door"
"oh the volume is too loud? ok... just press the left button and the music will go down..." (in this case i am aware that i will have to set through home assistant to set the panel to the media player entity)

what do you guys think?

**Comment by edwardtfn** [comment]
Not far from #1014

**Reply by TeDeVPrime** [reply]
yes but mine is better XD

**Reply by edwardtfn** [reply]
It's not a competition... I like to link similar things so we can see the full picture when looking for a solution. 😉
